src/osd/OSD.cc

```diff
@@ -5678,6 +5678,8 @@ void OSD::_preboot(epoch_t oldest, epoch_t newest)
   // if our map within recent history, try to add ourselves to the osdmap.
   if (osdmap->get_epoch() == 0) {
     derr << "waiting for initial osdmap" << dendl;
+  } else if (osdmap->is_destroyed(whoami)) {
+    derr << "osdmap says I am destroyed, waiting for re-creation" << dendl;
```

Comment:
It is probably better to actually exit here, since if we are recreated it will be a new osd with this id, with a new uuid, that is created.